lib/components/position_component.dart

```diff
@@ -25,26 +25,44 @@ import 'component.dart';
 /// They are translated by this component's (x,y). They do not need to fit
 /// within this component's (width, height).
 abstract class PositionComponent extends Component {
-  /// X position of this component on the screen (measured from the top left corner).
-  double x = 0.0;
+  /// The position of this component on the screen (relative to the anchor).
+  Vector2 position = Vector2.zero();
 
-  /// Y position of this component on the screen (measured from the top left corner).
-  double y = 0.0;
+  /// X position of this component on the screen (relative to the anchor).
+  double get x => position.x;
+  set x(double x) => position.x = x;
 
-  /// Angle (with respect to the x-axis) this component should be rendered with.
-  /// It is rotated around its anchor.
-  double angle = 0.0;
+  /// Y position of this component on the screen (relative to the anchor).
+  double get y => position.y;
+  set y(double y) => position.y = y;
+
+  /// The size that this component is rendered with.
+  /// This is not necessarily the source size of the asset.
+  Vector2 size = Vector2.zero();
 
   /// Width (size) that this component is rendered with.
-  /// This is not necessarily the source width of the asset.
-  double width = 0.0;
+  double get width => size.x;
+  set width(double width) => size.x = width;
 
   /// Height (size) that this component is rendered with.
-  /// This is not necessarily the source height of the asset.
-  double height = 0.0;
+  double get height => size.y;
+  set height(double height) => size.y = height;
+
+  /// Get the top left position regardless of the anchor
+  Vector2 get topLeftPosition => anchor.translate(position, size);
+
+  /// Set the top left position regardless of the anchor
+  set topLeftPosition(Vector2 position) {
+    this.position =
+        position + (anchor.relativePosition.clone()..multiply(size));
```

Comment:
nit: relativePosition creates a fresh copy each time, so I think you don't need to clone here